app/controllers/users_controller.rb

```diff
@@ -26,23 +26,25 @@ def create
     if @user.save
       #session[:user_id] = @user.id
       UserMailer.registration_confirmation(@user).deliver
-      flash[:success] = "Please confirm your email address to continue"
-      redirect_to home
+      #flash[:notice] = "Please confirm your email address to continue"
+      #flash.keep
+      redirect_to home_path, notice:"Please confirm your email address to continue"
     else
-      render :new
+      #flash[:notice] = "Ooooppss, something went wrong!"
+      render :new, notice: "Ooooppss, something went wrong!" 
     end
   end
 
   def confirm_email
     @user = User.find_by_confirm_token(params[:id])

```

Comment:
`@user = User.find_by(confirm_token: params[:id], email_activate: false)`
P.S. It would be better if you change `:id `to `:token` in `router.rb`